### PR TITLE
Allow server to treat unavailable revocation information as warnings

### PR DESCRIPTION
**Pull request by joelverhagen**
Server side allows temporarily unavailable revocation information. This setting is unused by client scenarios. Also, I flipped a oddly named bool to be more consistent.

Progress on https://github.com/NuGet/Engineering/issues/785.

The integration test is currently waiting on the test server @dtivel is working on. I have tested the code manually so I am fine with checking in the code without additional functional tests and coming back when the test server is ready. Let me know if you desire otherwise.

### CODE REVIEW

**Review by mishra14 — APPROVED** [review]
LGTM

**Review by emgarten — APPROVED** [review]

File: src/NuGet.Core/NuGet.Packaging/Signing/Verification/SignatureTrustAndValidityVerificationProvider.cs [hunk 1]
```diff
@@ -35,18 +35,36 @@ private PackageVerificationResult VerifyValidityAndTrust(Signature signature, Si
...
             return new SignedPackageVerificationResult(status, signature, issues);
         }
 
-        private Timestamp GetValidTimestamp(Signature signature, bool ignoreMultipleTimestamps, bool treatIssuesAsErrors, bool failIfNoTimestamp, List<SignatureLog> issues)
+        private Timestamp GetValidTimestamp(
+            Signature signature,
+            bool failIfMultipleTimestamps,
+            bool treatIssuesAsErrors,
+            bool failIfNoTimestamp,
+            bool failIfUnknownRevocation,
```
Comment (dtivel) [inline src/NuGet.Core/NuGet.Packaging/Signing/Verification/SignatureTrustAndValidityVerificationProvider.cs:73]: This parameter is unused.

File: src/NuGet.Core/NuGet.Packaging/Signing/Verification/SignatureTrustAndValidityVerificationProvider.cs [hunk 2]
```diff
@@ -187,7 +209,12 @@ private SignatureVerificationStatus VerifySignature(Signature signature, Timesta
         /// </summary>
         /// <param name="timestampCms">SignedCms response from the timestamp authority.</param>
         /// <param name="messageHash">Message hash that was signed and timestamped.</param>
-        private bool IsTimestampValid(Timestamp timestamp, byte[] messageHash, bool treatIssuesAsErrors, List<SignatureLog> issues)
+        private bool IsTimestampValid(
```
Comment (dtivel) [inline src/NuGet.Core/NuGet.Packaging/Signing/Verification/SignatureTrustAndValidityVerificationProvider.cs]: Remove or update XML doc comments.

**Review by dtivel — APPROVED** [review]
CC @PatoBeltran

File: src/NuGet.Core/NuGet.Packaging/Signing/Verification/SignatureTrustAndValidityVerificationProvider.cs (hunk 1 above)
Comment (joelverhagen) [reply src/NuGet.Core/NuGet.Packaging/Signing/Verification/SignatureTrustAndValidityVerificationProvider.cs:73]: Nice catch!

File: src/NuGet.Core/NuGet.Packaging/Signing/Verification/SignatureTrustAndValidityVerificationProvider.cs (hunk 2 above)
Comment (joelverhagen) [reply src/NuGet.Core/NuGet.Packaging/Signing/Verification/SignatureTrustAndValidityVerificationProvider.cs]: Fixed.

File: src/NuGet.Core/NuGet.Packaging/Signing/Verification/SignatureTrustAndValidityVerificationProvider.cs (hunk 1 above)
Comment (joelverhagen) [reply src/NuGet.Core/NuGet.Packaging/Signing/Verification/SignatureTrustAndValidityVerificationProvider.cs:73]: Fixed.

**Comment by joelverhagen** [comment]
I enabled the integration test by checking in a package that is already signed and that has unavailable CRL information.

**Review by PatoBeltran — APPROVED** [review]
💯

**Pull request merged**